spring-integration-amqp/src/test/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpointTests2.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.amqp.outbound;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.QueueBuilder;
+import org.springframework.amqp.core.QueueBuilder.Overflow;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.junit.RabbitAvailable;
+import org.springframework.amqp.rabbit.junit.RabbitAvailableCondition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.amqp.dsl.Amqp;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.messaging.MessageHandlingException;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.context.junit.jupiter.DisabledIf;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+/**
+ * @author Gary Russell
+ * @since 5.2
+ *
+ */
+@SpringJUnitConfig
+@RabbitAvailable(queues = "testConfirmOk")
+public class AmqpOutboundEndpointTests2 {
+
+	@Test
+	void testConfirmOk(@Autowired IntegrationFlow flow, @Autowired RabbitTemplate template) {
+		flow.getInputChannel().send(new GenericMessage<>("test", Collections.singletonMap("rk", "testConfirmOk")));
+		assertThat(template.receive("testConfirmOk")).isNotNull();
+	}
+
+	@Test
+	void testWithReturn(@Autowired IntegrationFlow flow) {
+		assertThatThrownBy(() -> flow.getInputChannel()
+				.send(new GenericMessage<>("test", Collections.singletonMap("rk", "junkjunk"))))
+						.isInstanceOf(MessageHandlingException.class)
+						.hasMessageStartingWith("Message was returned by the broker");
+	}
+
+	@Test
+	@DisabledIf("#{systemEnvironment['TRAVIS'] ?: false}") // needs RabbitMQ 3.7
```

Comment:
Is it really a thing? I mean doesn't it really come with not RabbitMQ 3.7 yet?
Can't we adjust the test somehow not having such a condition?

Thanks

Reply:
Well, we have to run travis on trusty for jdk reasons, and it still has 3.6.x (I checked the version in the travis log.

I don't see how we can adjust the test because it needs a 3.7 feature; I'd rather have the test work properly on our CI.

Reply:
OK. Let's live with this then! Eventually we will come back to clean code, when Travis supports 3.7.x already.